### PR TITLE
Use libsdl-org/setup-sdl GitHub action

### PR DESCRIPTION
**Pull request by madebr**
Looks like testautomation and testfile fail, on ci as well as on my machine.

Also, could anybody with an Apple look at the Macos timeouts?
The `libsdl-org/setup-sdl` action sets `DYLD_LIBRARY_PATH`.

### CODE REVIEW

**Comment by slouken** [comment]
I'm working on getting the SDL3 tests passing and then I'll take a look at the SDL2 tests, and can check it out on macOS

**Comment by madebr** [comment]
Thanks @sezero . The `testfile` test succeeds now.
However, in the mean time linux has regressed a bit: there is now a segmentation fault somewhere in the testautomation clipboard test.

Currently, by default, the libsdl-org/setup-sdl action does not install any dependencies, so there is no X11 backend.
Perhaps this is causing issues? (Should the default be changed?)

**Comment by sezero** [comment]
Don't know how setup-sdl works at all. However, we should run the testsuite selectively if possible: even `keyboard_getKeyFromScancode` of testautomation fails for me because of my keyboard layout. There also audio resample test failures among others.

**Comment by slouken** [comment]
@icculus and I can revisit the automated tests soon. Are these failing on main or sdl2-compat?

**Comment by madebr** [comment]
I think SDL3 main has issues. See #7466 for that.

**Comment by madebr** [comment]
To address https://github.com/libsdl-org/SDL/issues/8040, the sdl2-compat workflow need something as follows:
```
on:
  schedule:
    - cron: '5 1 * * *' # Build every day at 1:05am
```

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,31 +7,52 @@ jobs:
...
+        sudo apt-get install -y libgl1-mesa-dev libglu1-mesa-dev
+    - name: Configure (CMake)
+      shell: sh
+      run: |
+        cmake -B build -GNinja
+    - name: Build (CMake)
+      run: |
+        cmake --build build/
+    - name: Run build-time tests (CMake)
+      if: ${{ false }} # FIXME: enable build-time tests on CI
```
Comment (madebr) [inline .github/workflows/main.yml:47]: I left running the self tests disabled.
A follow-up PR will re-enable this line.

**Comment by madebr** [comment]
> To address [libsdl-org/SDL#8040](https://github.com/libsdl-org/SDL/issues/8040), the sdl2-compat workflow needs something as follows:
> 
> ```
> on:
>   schedule:
>     - cron: '5 1 * * *' # Build every day at 1:05am
> ```


Do we want this?

**Comment by slouken** [comment]
*deep breath*

Sure.

**Pull request merged**